### PR TITLE
Added ImageCompression Augmentation Function

### PR DESCRIPTION
**Pull request by denzel-datature**
Main changes added:

- Added imagecompression.py under src/discolight/augmentations/

### CODE REVIEW

File: src/discolight/augmentations/imagecompression.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import cv2
+from discolight.params.params import Params
+from .augmentation.types import ColorAugmentation, BoundedNumber
+from .decorators.accepts_probs import accepts_probs
+
+
+@accepts_probs
+class ImageCompression(ColorAugmentation):
+
+    """Image compress the given image.
```
Comment (ian-datature) [inline src/discolight/augmentations/imagecompression.py]:
```suggestion
    """Apply a compression effect to the given image.
```

File: src/discolight/augmentations/imagecompression.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    Works for both jpeg/jpg and png format.
```
Comment (ian-datature) [inline src/discolight/augmentations/imagecompression.py]: Not sure if this line is necessary. All augmentations should work for any image that can be loaded with OpenCV.

File: src/discolight/augmentations/imagecompression.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    This function is a lossy JPEG compression operation.
+    """
+
+    def __init__(self, strength):
+        """Construct a Image compression augmentation.
```
Comment (ian-datature) [inline src/discolight/augmentations/imagecompression.py]:
```suggestion
        """Construct a ImageCompression augmentation.
```

File: src/discolight/augmentations/imagecompression.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        """
+        super().__init__()
+        self.strength = strength
+
+    @staticmethod
+    def params():
+        """Return a Params object describing constructor parameters."""
+        return Params().add(
+            "strength",
+            "Compression strength between 0 to 100",
```
Comment (ian-datature) [inline src/discolight/augmentations/imagecompression.py]:
```suggestion
            "Compression strength",
```
Using the `BoundedNumber` type will cause information about the  bound to automatically appear in the documentation, so you don't need to add it to the param description.

**Review by ian-datature — CHANGES_REQUESTED** [review]

**Review by ian-datature — APPROVED** [review]

**Review by gkeechin — APPROVED** [review]

**Pull request merged**